inf-clojure.el

```diff
@@ -676,14 +676,22 @@ to continue it."
   "Remove subprompts from STRING."
   (replace-regexp-in-string inf-clojure-subprompt "" string))
 
+(defvar inf-clojure--previous-response-ended-with-prompt nil)
+
 (defun inf-clojure-preoutput-filter (str)
   "Preprocess the output STR from interactive commands."
   (inf-clojure--log-string str "<-RES----")
-  (cond
-   ((string-prefix-p "inf-clojure-" (symbol-name (or this-command last-command)))
-    ;; Remove subprompts and prepend a newline to the output string
-    (inf-clojure-chomp (concat "\n" (inf-clojure-remove-subprompts str))))
-   (t str)))
+  (let* ((last-line (car (last (split-string str "\n"))))
+         (response-was-one-line (string= last-line str))
+         (last-line-was-prompt (string-match inf-clojure-prompt last-line))
+         (filtered (if (and inf-clojure--previous-response-ended-with-prompt
+                            response-was-one-line
+                            last-line-was-prompt)
+                       ;; Prevent prompts from stacking in the same line.
+                       ""
+                     (inf-clojure-chomp (concat "\n" str)))))
```

Comment:
This has the effect of putting an extra empty line after manually entered forms (because enter adds \n) but I don't know any way to associate a response with a command.  I guess the other way to solve it would be to scrub the \n from the command after it's entered, not sure how practical that is.